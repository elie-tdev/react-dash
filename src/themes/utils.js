export const opacity = (hex, opacity) => {
  var result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
  if (!result) {
    throw new Error('Invalid hex', hex);
  }
  return `rgba(${parseInt(result[1], 16)}, ${parseInt(result[2], 16)}, ${parseInt(
    result[3],
    16,
  )}, ${opacity})`;
};

export const gradient = ({ type = 'linear', colors, background }) =>
  `${type}-gradient(${colors.join(', ')}), ${background};`;
