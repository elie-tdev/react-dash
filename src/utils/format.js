const formatter = new Intl.NumberFormat('en-US', {
  style: 'currency',
  currency: 'USD',
  minimumFractionDigits: 2,
});

const parseFloat = (str, val = 2) => {
  str = str && str.toString();
  str = str && str.indexOf('.') > -1 ? str.slice(0, str.indexOf('.') + val + 1) : str;
  return Number(str);
};

export const formatDollar = number => formatter.format(parseFloat(number, 2));
