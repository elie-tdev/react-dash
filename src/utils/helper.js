const maskAccountNumber = accNumber => {
  return accNumber && <>&bull;&bull;&bull;&bull; {accNumber.slice(accNumber.length - 4)}</>;
};

export default maskAccountNumber;
