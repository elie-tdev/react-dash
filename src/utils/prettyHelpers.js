export const prettyCurrency = (amount, currency = 'USD') => {
  if (amount == null) {
    return false;
  }
  if (amount < 0) {
    amount = 0;
  }
  return amount.toLocaleString('en', { style: 'currency', currency });
};

export const prettyDateTime = time => {
  if (time == null) {
    return false;
  }
  let d = new Date(time);
  return (
    d.toLocaleDateString('en', {
      month: '2-digit',
      day: '2-digit',
      year: '2-digit',
    }) +
    ' at ' +
    d.toLocaleTimeString('en', { hour: '2-digit', minute: '2-digit' }).replace(/^0(?:0:0?)?/, '')
  );
};

export const prettyDate = time => new Date(time).toLocaleDateString('en', {});

export const prettyTime = time =>
  new Date(time).toLocaleTimeString([], { hour: '2-digit', minute: '2-digit' });

export const prettyHour = time =>
  new Date(time)
    .toLocaleTimeString('en', { hour: '2-digit', minute: '2-digit' })
    .replace(/^0(?:0:0?)?/, '');

export const prettyPhone = number =>
  number.substr(2).replace(/(\d{1,2})(\d{1})?(\d{1,3})?(\d{1,4})?/, function (_, p1, p2, p3, p4) {
    let output = '';
    if (p1) output = `${p1}`;
    if (p2) output += `${p2}-`;
    if (p3) output += `${p3}-`;
    if (p4) output += `${p4}`;
    return output;
  });

export const minimumDays = (date, numDays) => {
  const d = new Date(date);
  const totalNumDays = daysIntoYear(d) + numDays;
  return Boolean(daysIntoYear(d) >= totalNumDays || totalNumDays <= 365);
};

export const daysIntoYear = date => {
  return (
    (Date.UTC(date.getFullYear(), date.getMonth(), date.getDate()) -
      Date.UTC(date.getFullYear(), 0, 0)) /
    24 /
    60 /
    60 /
    1000
  );
};

export const cleanAmountRegex = value => {
  return value.replace(/[^0-9]/g, '');
};

export const cleanAmount = value => {
  const pos = value.indexOf('.');
  let result;
  if (pos !== -1) {
    const part1 = value.substr(0, pos);
    const part2 = value.substr(pos + 1);
    result = cleanAmountRegex(part1) + '.' + cleanAmountRegex(part2);
  } else {
    result = cleanAmountRegex(value);
  }
  return result;
};
