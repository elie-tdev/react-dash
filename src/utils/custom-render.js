import React from 'react';
import { I18nextProvider } from 'react-i18next';
import { render } from '@testing-library/react';
import { CssBaseline } from '@material-ui/core';
import { BrowserRouter as Router } from 'react-router-dom';
import { ThemeProvider } from '@material-ui/core/styles';
import { MockedProvider } from '@apollo/client/testing';

import { MenuContextProvider } from '@/contexts';
import i18n from '@/translations';
import { LightTheme } from '@/themes';

const wrapper = ({ mocks = [], menuContextProviderProps = {} }) => {
  const Wrapper = ({ children }) => {
    return (
      <Router>
        <MockedProvider mocks={mocks} addTypename={false}>
          <ThemeProvider theme={LightTheme}>
            <MenuContextProvider {...menuContextProviderProps}>
              <CssBaseline />
              <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
            </MenuContextProvider>
          </ThemeProvider>
        </MockedProvider>
      </Router>
    );
  };

  return Wrapper;
};

const customRender = (ui, { mocks, menuContextProviderProps, ...restOptions } = {}) =>
  render(ui, { wrapper: wrapper({ mocks, menuContextProviderProps }), ...restOptions });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
