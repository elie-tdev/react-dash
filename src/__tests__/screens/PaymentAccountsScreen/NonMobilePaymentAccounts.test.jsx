import React from 'react';

import { render, fireEvent } from '@/utils/custom-render';
import NonMobilePaymentAccounts from '@/screens/PaymentAccountsScreen/NonMobilePaymentAccounts';

describe('NonMobilePaymentAccounts', () => {
  test('renders NonMobilePaymentAccounts', () => {
    const { container } = render(<NonMobilePaymentAccounts />);
    expect(container).toMatchSnapshot();
  });

  test('renders NonMobilePaymentAccounts with bank accounts', () => {
    const { container } = render(
      <NonMobilePaymentAccounts
        bankAccounts={[
          {
            routingNumber: '12341234',
            accountNumber: '123412341234',
            accountType: 'Checking',
            isPrimary: true,
          },
        ]}
        x
      />,
    );
    expect(container).toMatchSnapshot();
  });

  test('onClick of New Payment Account', async () => {
    const setIsNewPaymentAccountModalOpen = jest.fn();
    const handleClick = jest.fn();

    const { container, getAllByTestId } = render(
      <NonMobilePaymentAccounts
        onClick={handleClick()}
        bankAccounts={[
          {
            routingNumber: '12341234',
            accountNumber: '123412341234',
            accountType: 'Checking',
            isPrimary: true,
          },
        ]}
        setIsNewPaymentAccountModalOpen={setIsNewPaymentAccountModalOpen}
      />,
    );

    await new Promise(resolve => setTimeout(resolve, 0));
    const newPaymentButton = container.querySelector('button[data-testid="PrimaryButton"]');
    expect(await getAllByTestId(/PrimaryButton/i)).toHaveLength(1);
    await fireEvent.click(newPaymentButton);
    expect(handleClick).toHaveBeenCalled();
  });

  test('onClick of Edit Payment', async () => {
    const setEditingAccountIndex = jest.fn();
    const setIsEditPaymentAccountModalOpen = jest.fn();

    const handleClick = jest.fn();

    const { container, getAllByTestId } = render(
      <NonMobilePaymentAccounts
        onClick={handleClick()}
        bankAccounts={[
          {
            routingNumber: '12341234',
            accountNumber: '123412341234',
            accountType: 'Checking',
            isPrimary: true,
          },
        ]}
        setEditingAccountIndex={setEditingAccountIndex}
        setIsEditPaymentAccountModalOpen={setIsEditPaymentAccountModalOpen}
      />,
    );

    await new Promise(resolve => setTimeout(resolve, 0));
    const newPaymentButton = container.querySelector('button[data-testid="SecondaryButton"]');
    expect(await getAllByTestId(/SecondaryButton/i)).toHaveLength(1);
    await fireEvent.click(newPaymentButton);
    expect(handleClick).toHaveBeenCalled();
  });

  test('onClick of Delete', async () => {
    const setIsDeletionConfirmationModalOpenProp = jest.fn();
    const setEditingAccountIndex = jest.fn();
    const setIsEditPaymentAccountModalOpen = jest.fn();

    const handleClick = jest.fn();

    const { container, getAllByTestId } = render(
      <NonMobilePaymentAccounts
        onClick={handleClick()}
        bankAccounts={[
          {
            routingNumber: '12341234',
            accountNumber: '123412341234',
            accountType: 'Checking',
            isPrimary: false,
          },
        ]}
        setIsDeletionConfirmationModalOpenProp={setIsDeletionConfirmationModalOpenProp}
        setEditingAccountIndex={setEditingAccountIndex}
        setIsEditPaymentAccountModalOpen={setIsEditPaymentAccountModalOpen}
      />,
    );

    await new Promise(resolve => setTimeout(resolve, 0));
    const deleteButton = container.querySelector('button[data-testid="DangerButton"]');
    expect(await getAllByTestId(/DangerButton/i)).toHaveLength(1);
    await fireEvent.click(deleteButton);
    expect(handleClick).toHaveBeenCalled();
    expect(container).toMatchSnapshot();
  });

  test('visibility of account number', async () => {
    const handleClick = jest.fn();

    const { container } = render(
      <NonMobilePaymentAccounts
        onClick={handleClick()}
        bankAccounts={[
          {
            routingNumber: '12341234',
            accountNumber: '123412341234',
            accountType: 'Checking',
            isPrimary: false,
          },
        ]}
      />,
    );

    await new Promise(resolve => setTimeout(resolve, 0));
    const visibilityIcon = container.getElementsByTagName('svg');
    await expect(visibilityIcon).toHaveLength(1);

    await fireEvent.click(visibilityIcon[0]);
    await expect(handleClick).toHaveBeenCalled();
    expect(container).toMatchSnapshot();
  });

  test('invisibility of account number', async () => {
    jest.mock('react', () => ({
      ...jest.requireActual('react'),
      useState: jest
        .fn()
        .mockImplementation((isAccountNumberVisible = true, setIsAccountNumberVisible) => [
          isAccountNumberVisible,
          setIsAccountNumberVisible,
        ]),
    }));

    const handleClick = jest.fn();

    const { container } = render(
      <NonMobilePaymentAccounts
        onClick={handleClick()}
        bankAccounts={[
          {
            routingNumber: '12341234',
            accountNumber: '123412341234',
            accountType: 'Checking',
            isPrimary: false,
          },
        ]}
      />,
    );

    await new Promise(resolve => setTimeout(resolve, 0));
    const visibilityIcon = container.getElementsByTagName('svg');
    await expect(visibilityIcon).toHaveLength(1);
    await fireEvent.click(visibilityIcon[0]);
    await expect(handleClick).toHaveBeenCalled();
    expect(container).toMatchSnapshot();
  });
});
