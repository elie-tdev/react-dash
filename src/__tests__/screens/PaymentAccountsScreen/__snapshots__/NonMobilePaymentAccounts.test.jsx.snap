// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NonMobilePaymentAccounts invisibility of account number 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-cardBorder-337 MuiPaper-elevation0 MuiPaper-rounded"
    >
      <div
        class="MuiBox-root MuiBox-root-364 makeStyles-accountTypeBox-323"
      >
        <div
          class="MuiBox-root MuiBox-root-365"
        >
          <p
            class="MuiTypography-root makeStyles-accountType-324 makeStyles-text-331 MuiTypography-body1"
          >
            Checking
          </p>
           
          <p
            class="MuiTypography-root makeStyles-accountNumber-325 MuiTypography-body1"
          >
            •••• 
            1234
          </p>
        </div>
        <div
          class="MuiBox-root MuiBox-root-366"
        >
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
          >
            <span
              class="MuiSwitch-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-367 MuiSwitch-switchBase MuiSwitch-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-370 MuiSwitch-input"
                    type="checkbox"
                    value=""
                  />
                  <span
                    class="MuiSwitch-thumb"
                  />
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiSwitch-track"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Default
              </p>
            </span>
          </label>
        </div>
      </div>
      <hr
        class="MuiDivider-root makeStyles-divider-332"
      />
      <div
        class="MuiBox-root MuiBox-root-371 makeStyles-accountInfoBox-327"
      >
        <div
          class="MuiBox-root MuiBox-root-372 makeStyles-routingNumberBox-328"
        >
          <p
            class="MuiTypography-root makeStyles-text-331 MuiTypography-body1"
          >
            Routing Number
          </p>
          12341234
        </div>
        <div
          class="MuiBox-root MuiBox-root-373"
        >
          <p
            class="MuiTypography-root makeStyles-text-331 MuiTypography-body1"
          >
            Account Number
          </p>
          <div
            class="MuiBox-root MuiBox-root-374"
          >
            <p
              class="MuiTypography-root makeStyles-accountNumber-325 makeStyles-accountNumberMargin-326 MuiTypography-body1"
            >
              123412341234
            </p>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-visibilityIcon-333"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
              />
            </svg>
          </div>
        </div>
      </div>
      <hr
        class="MuiDivider-root makeStyles-divider-332"
      />
      <div
        class="MuiBox-root MuiBox-root-375 makeStyles-accountButtonsBox-330"
      >
        <button
          class="MuiButtonBase-root MuiButton-root makeStyles-root-376 makeStyles-root-377 MuiButton-contained makeStyles-noMarginButton-329 MuiButton-disableElevation"
          data-testid="SecondaryButton"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Edit Account
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root makeStyles-root-378 makeStyles-root-379 MuiButton-contained makeStyles-deleteButton-334 MuiButton-disableElevation"
          data-testid="DangerButton"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Delete
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
  >
    <div
      class="MuiBox-root MuiBox-root-380 makeStyles-newPaymentAccountBox-322"
    >
      <button
        class="MuiButtonBase-root MuiButton-root makeStyles-root-381 makeStyles-root-382 MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation"
        data-testid="PrimaryButton"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          + New Payment Account
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`NonMobilePaymentAccounts onClick of Delete 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-cardBorder-215 MuiPaper-elevation0 MuiPaper-rounded"
    >
      <div
        class="MuiBox-root MuiBox-root-242 makeStyles-accountTypeBox-201"
      >
        <div
          class="MuiBox-root MuiBox-root-243"
        >
          <p
            class="MuiTypography-root makeStyles-accountType-202 makeStyles-text-209 MuiTypography-body1"
          >
            Checking
          </p>
           
          <p
            class="MuiTypography-root makeStyles-accountNumber-203 MuiTypography-body1"
          >
            •••• 
            1234
          </p>
        </div>
        <div
          class="MuiBox-root MuiBox-root-244"
        >
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
          >
            <span
              class="MuiSwitch-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-245 MuiSwitch-switchBase MuiSwitch-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-248 MuiSwitch-input"
                    type="checkbox"
                    value=""
                  />
                  <span
                    class="MuiSwitch-thumb"
                  />
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiSwitch-track"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Default
              </p>
            </span>
          </label>
        </div>
      </div>
      <hr
        class="MuiDivider-root makeStyles-divider-210"
      />
      <div
        class="MuiBox-root MuiBox-root-249 makeStyles-accountInfoBox-205"
      >
        <div
          class="MuiBox-root MuiBox-root-250 makeStyles-routingNumberBox-206"
        >
          <p
            class="MuiTypography-root makeStyles-text-209 MuiTypography-body1"
          >
            Routing Number
          </p>
          12341234
        </div>
        <div
          class="MuiBox-root MuiBox-root-251"
        >
          <p
            class="MuiTypography-root makeStyles-text-209 MuiTypography-body1"
          >
            Account Number
          </p>
          <div
            class="MuiBox-root MuiBox-root-252"
          >
            <p
              class="MuiTypography-root makeStyles-accountNumber-203 makeStyles-accountNumberMargin-204 MuiTypography-body1"
            >
              •••• 
              1234
            </p>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-visibilityIcon-211"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
              />
            </svg>
          </div>
        </div>
      </div>
      <hr
        class="MuiDivider-root makeStyles-divider-210"
      />
      <div
        class="MuiBox-root MuiBox-root-253 makeStyles-accountButtonsBox-208"
      >
        <button
          class="MuiButtonBase-root MuiButton-root makeStyles-root-254 makeStyles-root-255 MuiButton-contained makeStyles-noMarginButton-207 MuiButton-disableElevation"
          data-testid="SecondaryButton"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Edit Account
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root makeStyles-root-256 makeStyles-root-257 MuiButton-contained makeStyles-deleteButton-212 MuiButton-disableElevation"
          data-testid="DangerButton"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Delete
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
  >
    <div
      class="MuiBox-root MuiBox-root-258 makeStyles-newPaymentAccountBox-200"
    >
      <button
        class="MuiButtonBase-root MuiButton-root makeStyles-root-259 makeStyles-root-260 MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation"
        data-testid="PrimaryButton"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          + New Payment Account
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`NonMobilePaymentAccounts renders NonMobilePaymentAccounts 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
  >
    <div
      class="MuiBox-root MuiBox-root-14 makeStyles-newPaymentAccountBox-1"
    >
      <button
        class="MuiButtonBase-root MuiButton-root makeStyles-root-15 makeStyles-root-16 MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation"
        data-testid="PrimaryButton"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          + New Payment Account
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`NonMobilePaymentAccounts renders NonMobilePaymentAccounts with bank accounts 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-cardBorder-32 MuiPaper-elevation0 MuiPaper-rounded"
    >
      <div
        class="MuiBox-root MuiBox-root-59 makeStyles-accountTypeBox-18"
      >
        <div
          class="MuiBox-root MuiBox-root-60"
        >
          <p
            class="MuiTypography-root makeStyles-accountType-19 makeStyles-text-26 MuiTypography-body1"
          >
            Checking
          </p>
           
          <p
            class="MuiTypography-root makeStyles-accountNumber-20 MuiTypography-body1"
          >
            •••• 
            1234
          </p>
        </div>
        <div
          class="MuiBox-root MuiBox-root-61"
        >
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
          >
            <span
              class="MuiSwitch-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-62 MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-checked-63 Mui-checked"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    checked=""
                    class="PrivateSwitchBase-input-65 MuiSwitch-input"
                    type="checkbox"
                    value=""
                  />
                  <span
                    class="MuiSwitch-thumb"
                  />
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiSwitch-track"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Default
              </p>
            </span>
          </label>
        </div>
      </div>
      <hr
        class="MuiDivider-root makeStyles-divider-27"
      />
      <div
        class="MuiBox-root MuiBox-root-66 makeStyles-accountInfoBox-22"
      >
        <div
          class="MuiBox-root MuiBox-root-67 makeStyles-routingNumberBox-23"
        >
          <p
            class="MuiTypography-root makeStyles-text-26 MuiTypography-body1"
          >
            Routing Number
          </p>
          12341234
        </div>
        <div
          class="MuiBox-root MuiBox-root-68"
        >
          <p
            class="MuiTypography-root makeStyles-text-26 MuiTypography-body1"
          >
            Account Number
          </p>
          <div
            class="MuiBox-root MuiBox-root-69"
          >
            <p
              class="MuiTypography-root makeStyles-accountNumber-20 makeStyles-accountNumberMargin-21 MuiTypography-body1"
            >
              •••• 
              1234
            </p>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-visibilityIcon-28"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
              />
            </svg>
          </div>
        </div>
      </div>
      <hr
        class="MuiDivider-root makeStyles-divider-27"
      />
      <div
        class="MuiBox-root MuiBox-root-70 makeStyles-accountButtonsBox-25"
      >
        <button
          class="MuiButtonBase-root MuiButton-root makeStyles-root-71 makeStyles-root-72 MuiButton-contained makeStyles-noMarginButton-24 MuiButton-disableElevation"
          data-testid="SecondaryButton"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Edit Account
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root makeStyles-root-73 makeStyles-root-74 MuiButton-contained makeStyles-deleteButton-29 MuiButton-disableElevation Mui-disabled Mui-disabled"
          data-testid="DangerButton"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Delete
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
  >
    <div
      class="MuiBox-root MuiBox-root-75 makeStyles-newPaymentAccountBox-17"
    >
      <button
        class="MuiButtonBase-root MuiButton-root makeStyles-root-76 makeStyles-root-77 MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation"
        data-testid="PrimaryButton"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          + New Payment Account
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`NonMobilePaymentAccounts visibility of account number 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-cardBorder-276 MuiPaper-elevation0 MuiPaper-rounded"
    >
      <div
        class="MuiBox-root MuiBox-root-303 makeStyles-accountTypeBox-262"
      >
        <div
          class="MuiBox-root MuiBox-root-304"
        >
          <p
            class="MuiTypography-root makeStyles-accountType-263 makeStyles-text-270 MuiTypography-body1"
          >
            Checking
          </p>
           
          <p
            class="MuiTypography-root makeStyles-accountNumber-264 MuiTypography-body1"
          >
            •••• 
            1234
          </p>
        </div>
        <div
          class="MuiBox-root MuiBox-root-305"
        >
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
          >
            <span
              class="MuiSwitch-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-306 MuiSwitch-switchBase MuiSwitch-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-309 MuiSwitch-input"
                    type="checkbox"
                    value=""
                  />
                  <span
                    class="MuiSwitch-thumb"
                  />
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiSwitch-track"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Default
              </p>
            </span>
          </label>
        </div>
      </div>
      <hr
        class="MuiDivider-root makeStyles-divider-271"
      />
      <div
        class="MuiBox-root MuiBox-root-310 makeStyles-accountInfoBox-266"
      >
        <div
          class="MuiBox-root MuiBox-root-311 makeStyles-routingNumberBox-267"
        >
          <p
            class="MuiTypography-root makeStyles-text-270 MuiTypography-body1"
          >
            Routing Number
          </p>
          12341234
        </div>
        <div
          class="MuiBox-root MuiBox-root-312"
        >
          <p
            class="MuiTypography-root makeStyles-text-270 MuiTypography-body1"
          >
            Account Number
          </p>
          <div
            class="MuiBox-root MuiBox-root-313"
          >
            <p
              class="MuiTypography-root makeStyles-accountNumber-264 makeStyles-accountNumberMargin-265 MuiTypography-body1"
            >
              123412341234
            </p>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-visibilityIcon-272"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
              />
            </svg>
          </div>
        </div>
      </div>
      <hr
        class="MuiDivider-root makeStyles-divider-271"
      />
      <div
        class="MuiBox-root MuiBox-root-314 makeStyles-accountButtonsBox-269"
      >
        <button
          class="MuiButtonBase-root MuiButton-root makeStyles-root-315 makeStyles-root-316 MuiButton-contained makeStyles-noMarginButton-268 MuiButton-disableElevation"
          data-testid="SecondaryButton"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Edit Account
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root makeStyles-root-317 makeStyles-root-318 MuiButton-contained makeStyles-deleteButton-273 MuiButton-disableElevation"
          data-testid="DangerButton"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Delete
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
  >
    <div
      class="MuiBox-root MuiBox-root-319 makeStyles-newPaymentAccountBox-261"
    >
      <button
        class="MuiButtonBase-root MuiButton-root makeStyles-root-320 makeStyles-root-321 MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation"
        data-testid="PrimaryButton"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          + New Payment Account
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
